fetch/fetch.go

```diff
@@ -482,18 +482,28 @@ func (f *Fetch) organizeRequests(requests []RequestMessage) map[p2p.Peer][][]Req
 		}
 		return nil
 	}
+	pm := map[p2p.Peer]struct{}{}
+	for _, p := range peers {
+		pm[p] = struct{}{}
+	}
```

Comment:
i think it is better to ask host.Connected or something like that. creating a map from peer list on every request is not ideal

Reply:
thanks. will add a method that checks fh.Network().Connectedness(p) == network.Connected